### ISSUE TITLE
a corner case of truncated response

### ISSUE THREAD

**Issue by sunbing81** [begriffs/postgrest#901]
Conditions to reproduce the bug:
1. a postgres function  returns a table;
2. the function returns only one row;
3. the table has a column named exactly the same as the function.

Steps to reproduce:

1. create a function:
```sql
create function public.test() returns table(test text, value int) language sql as $$
values ('hello', 1);
$$;
```

2. expected result:
```js
[{"test": "hello", "value": 1}]
```

3. the truncated response  of (```curl localhost:3000/rpc/test -XPOST -d''```):
```js
"hello"
```

Tested against both postgrest-0.4 (centos 6) and postgrest-0.4.2 (centos 7).

Thanks!

**Comment by begriffs** [comment]
Interesting, thanks for reporting. That's behavior is how functions used to work more generally, so this is probably some old logic bleeding through on the corner case.

**Comment by sunbing81** [comment]
that will be still ```"hello"```

**Comment by steve-chavez** [comment]
This was already fixed in #895 refactor commit, I've added a test case with the reported ```test() returns table(test text..``` function.
